src/analysis/inner_product_space/cross_product.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2022 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import analysis.inner_product_space.dual
+import analysis.inner_product_space.orientation
+
+/-! # The cross-product on an oriented real inner product space of dimension three
+
+This file gives a construction of the cross-product on an oriented real inner product space `E` of
+dimension three.
+
+## Implementation notes
+
+Notation `×₃` for `orientation.cross_product` should be defined locally in each file which uses it,
+since otherwise one would need a more cumbersome notation which mentions the orientation explicitly
+(something like `×₃[o]`).  Write
+```
+local infixl ` ×₃ `: 74 := o.cross_product
+```
+
+## TODO
+
+`matrix.cross_product` is a construction in co-ordinates of the cross-product on `R`^3, for `R` a
+commutative ring.  Either that construction should be superseded by this basis-free one (which still
+involves a loss of generality, in moving from a general commutative ring to just `ℝ`) or, if the two
+constructions will co-exist, there should be lemmas relating the two constructions.
+-/
+
+noncomputable theory
+
+open_locale real_inner_product_space
+open finite_dimensional
+
+local attribute [instance] fact_finite_dimensional_of_finrank_eq_succ
+
+variables {E : Type*} [inner_product_space ℝ E] [fact (finrank ℝ E = 3)]
```

Comment:
I worry about our growing usage of `fact` (for performance reasons, admittedly unquantifiable).

What do you think about introducing:
```lean
@[protect_proj] class finite_dimensional.finrank_eq
  (R V : Type*) [semiring R] [add_comm_monoid V] [module R V] (n : ℕ) :=
(finrank_eq : finite_dimensional.finrank R V = n)
```
since the pattern `[fact (finrank ...)]` occurs in a few places now and does seem useful?

(Incidentally I note that `finrank` inconsistently demands a `semiring` but a `add_comm_group` which I guess should be fixed :-/